// angular-confirmation-popover/dist/esm/src/confirmationPopoverWindow.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'angular-confirmation-popover/dist/esm/src/focus.directive';
import * as import3 from 'angular-confirmation-popover/dist/esm/src/confirmationPopoverWindow.component';
import * as import4 from 'angular-confirmation-popover/dist/esm/src/confirmationPopoverOptions.provider';
const styles_ConfirmationPopoverWindow:any[] = ['.popover[_ngcontent-%COMP%] {\n      display: block;\n    }'];
export const RenderType_ConfirmationPopoverWindow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_ConfirmationPopoverWindow,
  data: {}
}
);
function View_ConfirmationPopoverWindow_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'col-xs-6 col-6'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['col-xs-offset-3 col-offset-3']),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',([] as any[]),[
      [
        8,
        'className',
        0
      ]
      ,
      [
        8,
        'innerHTML',
        1
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>v.parent.context.options.onConfirm({clickEvent: $event})) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      import0.ɵdid(270336,(null as any),0,import2.Focus,[import0.ElementRef],{mwlFocus: [
        0,
        'mwlFocus'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'col-xs-6 col-6';
    const currVal_1:any = ck(v,2,0,v.parent.context.options.hideCancelButton);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_4:any = (v.parent.context.options.focusButton === 'confirm');
    ck(v,5,0,currVal_4);
  },(ck,v) => {
    const currVal_2:any = ('btn btn-block btn-' + v.parent.context.options.confirmButtonType);
    const currVal_3:any = v.parent.context.options.confirmText;
    ck(v,4,0,currVal_2,currVal_3);
  });
}
function View_ConfirmationPopoverWindow_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'col-xs-6 col-6'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['col-xs-offset-3 col-offset-3']),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',([] as any[]),[
      [
        8,
        'className',
        0
      ]
      ,
      [
        8,
        'innerHTML',
        1
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>v.parent.context.options.onCancel({clickEvent: $event})) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      import0.ɵdid(270336,(null as any),0,import2.Focus,[import0.ElementRef],{mwlFocus: [
        0,
        'mwlFocus'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'col-xs-6 col-6';
    const currVal_1:any = ck(v,2,0,v.parent.context.options.hideConfirmButton);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_4:any = (v.parent.context.options.focusButton === 'cancel');
    ck(v,5,0,currVal_4);
  },(ck,v) => {
    const currVal_2:any = ('btn btn-block btn-' + v.parent.context.options.cancelButtonType);
    const currVal_3:any = v.parent.context.options.cancelText;
    ck(v,4,0,currVal_2,currVal_3);
  });
}
function View_ConfirmationPopoverWindow_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),19,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'popover-arrow arrow'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'h3',[[
        'class',
        'popover-title'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'popover-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'p',([] as any[]),[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'row'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmationPopoverWindow_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmationPopoverWindow_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_3:boolean = !v.context.options.hideConfirmButton;
    ck(v,14,0,currVal_3);
    const currVal_4:boolean = !v.context.options.hideCancelButton;
    ck(v,17,0,currVal_4);
  },(ck,v) => {
    const currVal_0:any = ((((('popover ' + v.context.options.placement) + ' popover-') + v.context.options.placement) + ' ') + v.context.options.popoverClass);
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.options.title;
    ck(v,5,0,currVal_1);
    const currVal_2:any = v.context.options.message;
    ck(v,9,0,currVal_2);
  });
}
function View_ConfirmationPopoverWindow_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n    ']))],(null as any),(null as any));
}
export function View_ConfirmationPopoverWindow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵand(0,[[
        'defaultTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_ConfirmationPopoverWindow_1)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_ConfirmationPopoverWindow_4)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutletContext: [
        0,
        'ngTemplateOutletContext'
      ]
      ,
      ngTemplateOutlet: [
        1,
        'ngTemplateOutlet'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['options']),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import3.ConfirmationPopoverWindow = v.component;
    const currVal_0:any = ck(v,5,0,co.options);
    const currVal_1:any = (co.options.customTemplate || import0.ɵnov(v,1));
    ck(v,4,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ConfirmationPopoverWindow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ng-component',([] as any[]),(null as any),(null as any),(null as any),View_ConfirmationPopoverWindow_0,RenderType_ConfirmationPopoverWindow)),
    import0.ɵdid(2121728,(null as any),0,import3.ConfirmationPopoverWindow,[import4.ConfirmationPopoverWindowOptions],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ConfirmationPopoverWindowNgFactory:import0.ComponentFactory<import3.ConfirmationPopoverWindow> = import0.ɵccf('ng-component',import3.ConfirmationPopoverWindow,View_ConfirmationPopoverWindow_Host_0,{},{},([] as any[]));
