// angular-confirmation-popover/dist/esm/src/confirmationPopoverWindow.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'angular-confirmation-popover/dist/esm/src/confirmationPopoverWindow.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/constants';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from 'angular-confirmation-popover/dist/esm/src/confirmationPopoverOptions.provider';
import * as import9 from '../../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import10 from './focus.directive.ngfactory';
import * as import11 from '@angular/core/src/linker/view_container';
import * as import12 from '@angular/core/src/change_detection/change_detection_util';
import * as import13 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import14 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from 'angular-confirmation-popover/dist/esm/src/focus.directive';
import * as import17 from '@angular/common/src/directives/ng_class';
import * as import18 from '@angular/core/src/security';
import * as import19 from '../../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import20 from '@angular/core/src/linker/template_ref';
import * as import21 from '@angular/common/src/directives/ng_if';
import * as import22 from '../../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import23 from '@angular/common/src/directives/ng_template_outlet';
export class Wrapper_ConfirmationPopoverWindow {
  /*private*/ _eventHandler:Function;
  context:import0.ConfirmationPopoverWindow;
  /*private*/ _changed:boolean;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.ConfirmationPopoverWindow(p0);
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_ConfirmationPopoverWindow_Host:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,([] as any[]),{});
class View_ConfirmationPopoverWindow_Host0 extends import1.AppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.ConfirmationPopoverWindow>;
  _ConfirmationPopoverWindow_0_3:Wrapper_ConfirmationPopoverWindow;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_ConfirmationPopoverWindow_Host0,renderType_ConfirmationPopoverWindow_Host,import5.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'ng-component',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_ConfirmationPopoverWindow0(this.viewUtils,this,0,this._el_0);
    this._ConfirmationPopoverWindow_0_3 = new Wrapper_ConfirmationPopoverWindow(this.injectorGet(import8.ConfirmationPopoverWindowOptions,this.parentIndex));
    this.compView_0.create(this._ConfirmationPopoverWindow_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._ConfirmationPopoverWindow_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.ConfirmationPopoverWindow) && (0 === requestNodeIndex))) { return this._ConfirmationPopoverWindow_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._ConfirmationPopoverWindow_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._ConfirmationPopoverWindow_0_3.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const ConfirmationPopoverWindowNgFactory:import7.ComponentFactory<import0.ConfirmationPopoverWindow> = new import7.ComponentFactory<import0.ConfirmationPopoverWindow>('ng-component',View_ConfirmationPopoverWindow_Host0,import0.ConfirmationPopoverWindow);
const styles_ConfirmationPopoverWindow:any[] = ['.popover[_ngcontent-%COMP%] {\n      display: block;\n    }'];
class View_ConfirmationPopoverWindow2 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import9.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _Focus_2_3:import10.Wrapper_Focus;
  _text_3:any;
  _text_4:any;
  _map_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import11.ViewContainer) {
    super(View_ConfirmationPopoverWindow2,renderType_ConfirmationPopoverWindow,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_7 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'col-xs-offset-3 col-offset-3': p0};
    });
    this._expr_8 = import12.UNINITIALIZED;
    this._expr_9 = import12.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','col-xs-6 col-6'),(null as any));
    this._NgClass_0_3 = new import9.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n              ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'button',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._Focus_2_3 = new import10.Wrapper_Focus(this.renderer,new import15.ElementRef(this._el_2));
    this._text_3 = this.renderer.createText(this._el_2,'\n              ',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_2,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.Focus) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._Focus_2_3.context; }
    if (((token === import17.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'col-xs-6 col-6';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_7(this.parentView.context.options.hideCancelButton);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.parentView.context.options.focusButton === 'confirm');
    this._Focus_2_3.check_mwlFocus(currVal_2_0_0,throwOnChange,false);
    this._Focus_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_8:any = ('btn btn-block btn-' + this.parentView.context.options.confirmButtonType);
    if (import3.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementProperty(this._el_2,'className',currVal_8);
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.parentView.context.options.confirmText;
    if (import3.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import18.SecurityContext.HTML,currVal_9));
      this._expr_9 = currVal_9;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.options.onConfirm()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_ConfirmationPopoverWindow3 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import9.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _Focus_2_3:import10.Wrapper_Focus;
  _text_3:any;
  _text_4:any;
  _map_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import11.ViewContainer) {
    super(View_ConfirmationPopoverWindow3,renderType_ConfirmationPopoverWindow,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_7 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'col-xs-offset-3 col-offset-3': p0};
    });
    this._expr_8 = import12.UNINITIALIZED;
    this._expr_9 = import12.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','col-xs-6 col-6'),(null as any));
    this._NgClass_0_3 = new import9.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n              ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'button',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._Focus_2_3 = new import10.Wrapper_Focus(this.renderer,new import15.ElementRef(this._el_2));
    this._text_3 = this.renderer.createText(this._el_2,'\n              ',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_2,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.Focus) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._Focus_2_3.context; }
    if (((token === import17.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'col-xs-6 col-6';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_7(this.parentView.context.options.hideConfirmButton);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.parentView.context.options.focusButton === 'cancel');
    this._Focus_2_3.check_mwlFocus(currVal_2_0_0,throwOnChange,false);
    this._Focus_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_8:any = ('btn btn-block btn-' + this.parentView.context.options.cancelButtonType);
    if (import3.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementProperty(this._el_2,'className',currVal_8);
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.parentView.context.options.cancelText;
    if (import3.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import18.SecurityContext.HTML,currVal_9));
      this._expr_9 = currVal_9;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.options.onCancel()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_ConfirmationPopoverWindow1 extends import1.AppView<any> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _el_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _anchor_13:any;
  /*private*/ _vc_13:import11.ViewContainer;
  _TemplateRef_13_5:any;
  _NgIf_13_6:import19.Wrapper_NgIf;
  _text_14:any;
  _anchor_15:any;
  /*private*/ _vc_15:import11.ViewContainer;
  _TemplateRef_15_5:any;
  _NgIf_15_6:import19.Wrapper_NgIf;
  _text_16:any;
  _text_17:any;
  _text_18:any;
  _text_19:any;
  /*private*/ _expr_26:any;
  /*private*/ _expr_27:any;
  /*private*/ _expr_28:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import11.ViewContainer) {
    super(View_ConfirmationPopoverWindow1,renderType_ConfirmationPopoverWindow,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_26 = import12.UNINITIALIZED;
    this._expr_27 = import12.UNINITIALIZED;
    this._expr_28 = import12.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n      ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,(null as any),'div',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray2(2,'class','popover-arrow arrow'),(null as any));
    this._text_4 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_1,'h3',new import3.InlineArray2(2,'class','popover-title'),(null as any));
    this._text_6 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._el_7 = import3.createRenderElement(this.renderer,this._el_1,'div',new import3.InlineArray2(2,'class','popover-content'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'\n          ',(null as any));
    this._el_9 = import3.createRenderElement(this.renderer,this._el_7,'p',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_10 = this.renderer.createText(this._el_7,'\n          ',(null as any));
    this._el_11 = import3.createRenderElement(this.renderer,this._el_7,'div',new import3.InlineArray2(2,'class','row'),(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'\n            ',(null as any));
    this._anchor_13 = this.renderer.createTemplateAnchor(this._el_11,(null as any));
    this._vc_13 = new import11.ViewContainer(13,11,this,this._anchor_13);
    this._TemplateRef_13_5 = new import20.TemplateRef_(this,13,this._anchor_13);
    this._NgIf_13_6 = new import19.Wrapper_NgIf(this._vc_13.vcRef,this._TemplateRef_13_5);
    this._text_14 = this.renderer.createText(this._el_11,'\n            ',(null as any));
    this._anchor_15 = this.renderer.createTemplateAnchor(this._el_11,(null as any));
    this._vc_15 = new import11.ViewContainer(15,11,this,this._anchor_15);
    this._TemplateRef_15_5 = new import20.TemplateRef_(this,15,this._anchor_15);
    this._NgIf_15_6 = new import19.Wrapper_NgIf(this._vc_15.vcRef,this._TemplateRef_15_5);
    this._text_16 = this.renderer.createText(this._el_11,'\n          ',(null as any));
    this._text_17 = this.renderer.createText(this._el_7,'\n        ',(null as any));
    this._text_18 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._text_19 = this.renderer.createText((null as any),'\n    ',(null as any));
    this.init(this._text_19,((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._anchor_13,
      this._text_14,
      this._anchor_15,
      this._text_16,
      this._text_17,
      this._text_18,
      this._text_19
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.TemplateRef) && (13 === requestNodeIndex))) { return this._TemplateRef_13_5; }
    if (((token === import21.NgIf) && (13 === requestNodeIndex))) { return this._NgIf_13_6.context; }
    if (((token === import20.TemplateRef) && (15 === requestNodeIndex))) { return this._TemplateRef_15_5; }
    if (((token === import21.NgIf) && (15 === requestNodeIndex))) { return this._NgIf_15_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_13_0_0:boolean = !this.context.options.hideConfirmButton;
    this._NgIf_13_6.check_ngIf(currVal_13_0_0,throwOnChange,false);
    this._NgIf_13_6.ngDoCheck(this,this._anchor_13,throwOnChange);
    const currVal_15_0_0:boolean = !this.context.options.hideCancelButton;
    this._NgIf_15_6.check_ngIf(currVal_15_0_0,throwOnChange,false);
    this._NgIf_15_6.ngDoCheck(this,this._anchor_15,throwOnChange);
    this._vc_13.detectChangesInNestedViews(throwOnChange);
    this._vc_15.detectChangesInNestedViews(throwOnChange);
    const currVal_26:any = ((((('popover ' + this.context.options.placement) + ' popover-') + this.context.options.placement) + ' ') + this.context.options.popoverClass);
    if (import3.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this.renderer.setElementProperty(this._el_1,'className',currVal_26);
      this._expr_26 = currVal_26;
    }
    const currVal_27:any = this.context.options.title;
    if (import3.checkBinding(throwOnChange,this._expr_27,currVal_27)) {
      this.renderer.setElementProperty(this._el_5,'innerHTML',this.viewUtils.sanitizer.sanitize(import18.SecurityContext.HTML,currVal_27));
      this._expr_27 = currVal_27;
    }
    const currVal_28:any = this.context.options.message;
    if (import3.checkBinding(throwOnChange,this._expr_28,currVal_28)) {
      this.renderer.setElementProperty(this._el_9,'innerHTML',this.viewUtils.sanitizer.sanitize(import18.SecurityContext.HTML,currVal_28));
      this._expr_28 = currVal_28;
    }
  }
  destroyInternal():void {
    this._vc_13.destroyNestedViews();
    this._vc_15.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
    cb(this._el_1,ctx);
    cb(this._text_19,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 13)) { return new View_ConfirmationPopoverWindow2(this.viewUtils,this,13,this._anchor_13,this._vc_13); }
    if ((nodeIndex == 15)) { return new View_ConfirmationPopoverWindow3(this.viewUtils,this,15,this._anchor_15,this._vc_15); }
    return (null as any);
  }
}
class View_ConfirmationPopoverWindow4 extends import1.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import11.ViewContainer) {
    super(View_ConfirmationPopoverWindow4,renderType_ConfirmationPopoverWindow,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n    ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
var renderType_ConfirmationPopoverWindow:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.Emulated,styles_ConfirmationPopoverWindow,{});
export class View_ConfirmationPopoverWindow0 extends import1.AppView<import0.ConfirmationPopoverWindow> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import11.ViewContainer;
  _TemplateRef_1_4:any;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _vc_3:import11.ViewContainer;
  _TemplateRef_3_5:any;
  _NgTemplateOutlet_3_6:import22.Wrapper_NgTemplateOutlet;
  _text_4:any;
  _map_10:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_ConfirmationPopoverWindow0,renderType_ConfirmationPopoverWindow,import5.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
    this._map_10 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {options: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import11.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_4 = new import20.TemplateRef_(this,1,this._anchor_1);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_3 = new import11.ViewContainer(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import20.TemplateRef_(this,3,this._anchor_3);
    this._NgTemplateOutlet_3_6 = new import22.Wrapper_NgTemplateOutlet(this._vc_3.vcRef);
    this._text_4 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2,
      this._anchor_3,
      this._text_4
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import20.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_4; }
    if (((token === import20.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import23.NgTemplateOutlet) && (3 === requestNodeIndex))) { return this._NgTemplateOutlet_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = this._map_10(this.context.options);
    this._NgTemplateOutlet_3_6.check_ngOutletContext(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = (this.context.options.customTemplate || this._TemplateRef_1_4);
    this._NgTemplateOutlet_3_6.check_ngTemplateOutlet(currVal_3_0_1,throwOnChange,false);
    this._NgTemplateOutlet_3_6.ngDoCheck(this,this._anchor_3,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
    this._vc_3.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
    this._vc_3.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 1)) { return new View_ConfirmationPopoverWindow1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    if ((nodeIndex == 3)) { return new View_ConfirmationPopoverWindow4(this.viewUtils,this,3,this._anchor_3,this._vc_3); }
    return (null as any);
  }
}
