// ngx-swiper-wrapper/dist/lib/swiper.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './swiper.component.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from 'ngx-swiper-wrapper/dist/lib/swiper.component';
import * as import3 from 'ngx-swiper-wrapper/dist/lib/swiper.interfaces';
const styles_SwiperComponent:any[] = [import0.styles];
export const RenderType_SwiperComponent:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_SwiperComponent,
  data: {}
}
);
export function View_SwiperComponent_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    import1.ɵqud(201326592,1,{swiperWrapper: 0}),
      (l()(),import1.ɵeld(0,(null as any),(null as any),14,'div',[[
        'class',
        'swiper-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n  '])),
    (l()(),import1.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'swiperWrapper',
        1
      ]

    ]
      ,(null as any),3,'div',[[
        'class',
        'swiper-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n    '])),
    import1.ɵncd((null as any),0),
    (l()(),import1.ɵted((null as any),['\n  '])),
    (l()(),import1.ɵted((null as any),['\n\n  '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'swiper-scrollbar'
      ]
      ],[[
        8,
        'hidden',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n\n  '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'swiper-button-prev'
      ]
    ],[
      [
        8,
        'hidden',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n  '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'swiper-button-next'
      ]
    ],[
      [
        8,
        'hidden',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n\n  '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'swiper-pagination'
      ]
      ],[[
        8,
        'hidden',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.SwiperComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onIndexSelect($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['\n'])),
    (l()(),import1.ɵted((null as any),['\n']))
  ]
  ,(null as any),(ck,v) => {
    var co:import2.SwiperComponent = v.component;
    const currVal_0:boolean = !co.showScrollbar;
    ck(v,8,0,currVal_0);
    const currVal_1:boolean = !co.showButtons;
    const currVal_2:any = co.isAtFirst;
    ck(v,10,0,currVal_1,currVal_2);
    const currVal_3:boolean = !co.showButtons;
    const currVal_4:any = co.isAtLast;
    ck(v,12,0,currVal_3,currVal_4);
    const currVal_5:boolean = !co.showPagination;
    ck(v,14,0,currVal_5);
  });
}
function View_SwiperComponent_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    (l()(),import1.ɵeld(0,(null as any),(null as any),1,'swiper',([] as any[]),[
      [
        8,
        'hidden',
        0
      ]
      ,
      [
        2,
        'swiper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_SwiperComponent_0,RenderType_SwiperComponent)),
    import1.ɵdid(516096,(null as any),0,import2.SwiperComponent,[
      import1.NgZone,
      import1.ElementRef,
      import1.KeyValueDiffers,
      [
        2,
        import3.SwiperConfig
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,1).hidden;
    const currVal_1:any = import1.ɵnov(v,1).useSwiperClass;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const SwiperComponentNgFactory:import1.ComponentFactory<import2.SwiperComponent> = import1.ɵccf('swiper',import2.SwiperComponent,View_SwiperComponent_Host_0,{
  hidden: 'hidden',
  disabled: 'disabled',
  config: 'config',
  runInsideAngular: 'runInsideAngular',
  useSwiperClass: 'useSwiperClass'
}
,{
  indexChange: 'indexChange',
  s_init: 'init',
  s_slideChangeStart: 'slideChangeStart',
  s_slideChangeEnd: 'slideChangeEnd',
  s_slideNextStart: 'slideNextStart',
  s_slideNextEnd: 'slideNextEnd',
  s_slidePrevStart: 'slidePrevStart',
  s_slidePrevEnd: 'slidePrevEnd',
  s_transitionStart: 'transitionStart',
  s_transitionEnd: 'transitionEnd',
  s_touchStart: 'touchStart',
  s_touchMove: 'touchMove',
  s_touchMoveOpposite: 'touchMoveOpposite',
  s_sliderMove: 'sliderMove',
  s_touchEnd: 'touchEnd',
  s_click: 'click',
  s_tap: 'tap',
  s_doubleTap: 'doubleTap',
  s_imagesReady: 'imagesReady',
  s_progress: 'progress',
  s_reachBeginning: 'reachBeginning',
  s_reachEnd: 'reachEnd',
  s_destroy: 'destroy',
  s_setTranslate: 'setTranslate',
  s_setTransition: 'setTransition',
  s_autoplay: 'autoplay',
  s_autoplayStart: 'autoplayStart',
  s_autoplayStop: 'autoplayStop',
  s_lazyImageLoad: 'lazyImageLoad',
  s_lazyImageReady: 'lazyImageReady',
  s_paginationRendered: 'paginationRendered',
  s_scroll: 'scroll'
}
,['*']);
